mlflow/projects/__init__.py

```diff
@@ -406,31 +356,32 @@ def _run_project(project, entry_point, work_dir, parameters, use_conda, storage_
         commands.append("source activate %s" % _get_conda_env_name(conda_env_path))
 
     # Create a new run and log every provided parameter into it.
-    active_run = tracking.start_run(experiment_id=experiment_id,
-                                    source_name=project.uri,
-                                    source_version=tracking._get_git_commit(work_dir),
-                                    entry_point_name=entry_point,
-                                    source_type=SourceType.PROJECT)
+    exp_id_for_run = experiment_id or tracking._get_experiment_id()
+    assert tracking._active_run is None
+    active_run = tracking._create_run(
+        experiment_id=exp_id_for_run, source_name=project.uri,
+        source_version=tracking._get_git_commit(work_dir), entry_point_name=entry_point,
+        source_type=SourceType.PROJECT)
+    assert tracking._active_run is None
     if parameters is not None:
         for key, value in parameters.items():
             active_run.log_param(Param(key, value))
     # Add the run id into a magic environment variable that the subprocess will read,
     # causing it to reuse the run.
     exp_id = experiment_id or tracking._get_experiment_id()
     env_map = {
-        tracking._RUN_NAME_ENV_VAR: active_run.run_info.run_uuid,
```

Comment:
This looks like a backwards- and forwards-incompatible change, f we assume that these environment variables are passed to a Spark cluster, for example. Do we care? Is there ever a situation in which the local mlflow and the Databricks mlflow can be of different versions?

Reply:
It's a good point, I think we might not care - by default our logic for running Databricks runs installs the same MLflow version on Databricks as that of the local client (i.e. if you're running a Databricks run with MLflow x.y.z, we'll install MLflow x.y.z on the cluster & use that when executing `mlflow run`). I can't think of a case where a user wouldn't want this to be the case, especially since MLflow is only used in the cluster to invoke `mlflow run http://user/project/uri`, and the user's project can depend on whatever MLflow version it wants in its conda.yaml.

The one tricky case I can think of is integration testing Databricks runs - for that we could do something like publish a nightly MLflow build & run tests using that version locally & on the Databricks cluster.